### PR TITLE
incomplete label to daily files

### PR DESCRIPTION
**Pull request by nicHoch**

### CODE REVIEW

File: stixcore/processing/pipeline_cli.py [hunk 1]
```diff
@@ -170,7 +170,7 @@ def main():
         spicemeta = Path(args.spice_file)
     else:
         _spm = SpiceKernelManager(Path(CONFIG.get('Paths', 'spice_kernels')))
-        spicemeta = _spm.get_latest_mk(top_n=30)
+        spicemeta = _spm.get_latest_mk(top_n=10)
```
Comment (samaloney) [inline stixcore/processing/pipeline_cli.py:173]: why load more the the last one?

File: stixcore/io/fits/processors.py [hunk 2]
```diff
@@ -91,8 +91,12 @@ def generate_filename(cls, product, *, version, date_range, status=''):
         if 'tc_packet_seq_control' in product.control.colnames and user_req != '':
             tc_control = f'-{product.control["tc_packet_seq_control"][0]:05d}'
 
+        incomplete = ''
+        if not header and product.level != 'LB' and product.fits_daily_file is True:
+            incomplete = 'U'
+
         return f'solo_{product.level}_stix-{product.type}-{product.name.replace("_", "-")}' \
-               f'_{date_range}_V{version:02d}{status}{user_req}{tc_control}.fits'
+               f'_{date_range}_V{version:02d}{incomplete}{status}{user_req}{tc_control}.fits'
```
Comment (samaloney) [inline stixcore/io/fits/processors.py:99]: This was the idea of the status flag so status could be `C`, `I` or `U` why add another variable?

File: stixcore/io/fits/processors.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ class FitsProcessor:
     # TODO abstract some general processing pattern methods
 
     @classmethod
-    def generate_filename(cls, product, *, version, date_range, status=''):
+    def generate_filename(cls, product, *, version, date_range, status='', header=True):
```
Comment (samaloney) [inline stixcore/io/fits/processors.py:67]: What does `header=True` mean also needs to be added to the doc string

File: stixcore/io/fits/processors.py [hunk 4]
```diff
@@ -499,7 +505,7 @@ def write_fits(self, product, path=None, *, version=1):
                                   for version, range in prod.idb_versions.items()],
                                   names=["version", "obt_start", "obt_end"])
 
-            primary_header = self.generate_primary_header(filename, prod, version=version)
+            primary_header = self.generate_primary_header(filename_header, prod, version=version)
```
Comment (samaloney) [inline stixcore/io/fits/processors.py:508]: So the name of the file and the filename header will be different?

File: stixcore/io/fits/processors.py [hunk 5]
```diff
@@ -787,7 +794,7 @@ def write_fits(self, product, *, version=1):
                                   for version, range in prod.idb_versions.items()],
                                   names=["version", "obt_start", "obt_end"])
 
-            primary_header, header_override = self.generate_primary_header(filename, prod,
+            primary_header, header_override = self.generate_primary_header(filename_header, prod,
```
Comment (samaloney) [inline stixcore/io/fits/processors.py:797]: So the name of the file and the filename header will be different?

File: stixcore/processing/publish.py [hunk 6]
```diff
@@ -349,6 +352,116 @@ def send_mail_report(files):
...
+            for hdu in f:
+                now = datetime.now().isoformat(timespec='milliseconds')
+                hdu.header['HISTORY'] = f'updated ephemeris header with latest kernel at {now}'
+                hdu.header.update(ephemeris_headers)
+                # just update the first prima HDU
+                break
+        logger.info(f"updated ephemeris headers of {fits_file}")
+
+
+def addBSDComment(p, rid_lut):
```
Comment (samaloney) [inline stixcore/processing/publish.py:371]:
```suggestion
def add_BSD_comment(p, rid_lut):
```
Also add short docstring, I guess from the context rid is request but not sure

File: stixcore/processing/publish.py [hunk 7]
```diff
@@ -349,6 +352,116 @@ def send_mail_report(files):
...
+            c_entry = f"BSD request id: '{rid}' reason: '{reason}'"
+            fits.setval(p, "COMMENT", value=c_entry)
+            return c_entry
+    except Exception as e:
+        logger.debug(e, exc_info=True)
+        logger.info(f"no BSD request comment added nothing found in LUT for rid: {rid}")
+    return ''
+
+
+def addHistory(p, name):
```
Comment (samaloney) [inline stixcore/processing/publish.py:388]:
```suggestion
def add_history(p, name):
```

Also add short docstring

File: stixcore/processing/publish.py [hunk 8]
```diff
@@ -349,6 +352,116 @@ def send_mail_report(files):
...
+    time_formated = datetime.now().isoformat(timespec='milliseconds')
+    h_entry = f"published to ESA SOAR as '{name}' on {time_formated}"
+    fits.setval(p, "HISTORY", value=h_entry)
+
+
+def copyFile(scp, p, target_dir, add_history=True):
```
Comment (samaloney) [inline stixcore/processing/publish.py:394]:
```suggestion
def copy_file(scp, p, target_dir, add_history=True):
```

Also add short docstring

File: stixcore/processing/publish.py [hunk 9]
```diff
@@ -349,6 +352,116 @@ def send_mail_report(files):
...
+        if scp:
+            scp.put(p, remote_path=target_dir)
+        else:
+            shutil.copy(p, target_dir)
+        return (True, None)
+    except Exception as e:
+        return (False, e)
+
+
+def read_rid_lut(file, update=False):
```
Comment (samaloney) [inline stixcore/processing/publish.py:407]: Short docstring what does this do?

File: stixcore/products/level0/housekeepingL0.py [hunk 10]
```diff
@@ -39,6 +39,10 @@ def __init__(self, *, service_type, service_subtype, ssid, control, data,
         self.level = 'L0'
         self.type = 'hk'
 
+    @property
+    def fits_daily_file(self):
```
Comment (samaloney) [inline stixcore/products/level0/housekeepingL0.py:43]: I haven't seen this referenced/used?

**Review by samaloney — COMMENTED** [review]
Its a complex change so hard to tell but looks good in general.

The `status` variable was include for the purpose of indicating the completeness of the files could this be used?

Are the L1 and L2 files only generated for complete files?

File: stixcore/io/fits/processors.py (hunk 2 above)
Comment (nicHoch) [reply stixcore/io/fits/processors.py:99]: wow did not realized that this was already foreseen a time ago
will use the status

File: stixcore/io/fits/processors.py (hunk 4 above)
Comment (nicHoch) [reply stixcore/io/fits/processors.py:508]: s.a.:

File: stixcore/processing/publish.py (hunk 7 above)
Comment (nicHoch) [reply stixcore/processing/publish.py]: done

File: stixcore/processing/pipeline_cli.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nicHoch) [reply stixcore/processing/pipeline_cli.py:173]: this is only used in the CLI sometimes a kernel was corrupt the top_n tells that it should fall back to the previes one if an error happens. but not further as top_n meta kernel files

File: stixcore/io/fits/processors.py (hunk 4 above)
Comment (nicHoch) [reply stixcore/io/fits/processors.py:508]: updated so the the FILENAME keyword also has the same value as the filename. will be updated later again

File: stixcore/io/fits/processors.py (hunk 3 above)
Comment (samaloney) [reply stixcore/io/fits/processors.py:67]: I don't see header in the doc string?

**Review by samaloney — APPROVED** [review]
LGTM - I don't see it but I guess the filename and filename header keyword will match?

**Review by samaloney — APPROVED** [review]

File: stixcore/io/fits/processors.py [hunk 11]
```diff
@@ -775,12 +779,20 @@ def write_fits(self, product, *, version=1):
...
                 existing = Product(fitspath)
                 logger.debug('Existing %s, Current %s', existing, prod)
                 prod = prod + existing
                 logger.debug('Combined %s', prod)
+            elif fitspath_complete.exists():
+                logger.info('Complete Fits file %s exists appending data', fitspath.name)
+                existing = Product(fitspath_complete)
+                logger.debug('Existing %s, Current %s', existing, prod)
+                prod = prod + existing
+                logger.debug('Combined %s', prod)
```
[the remark this answers is not in the document]
Comment (nicHoch) [reply stixcore/io/fits/processors.py:795]: this is the trick for the "cutover" path.
Daily files will new always be incomplete so the only "finding a complete" file is during cutover

**Pull request merged**